### PR TITLE
ucd-parse: Add support for parsing ArabicShaping.txt

### PR DESCRIPTION
**Pull request by wezm**

### CODE REVIEW

File: ucd-parse/src/arabic_shaping.rs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    pub codepoint: Codepoint,
+    /// A short schematic name for the codepoint.
+    ///
+    /// The schematic name is descriptive of the shape, based as consistently as
+    /// possible on a name for the skeleton and then the diacritic marks applied
+    /// to the skeleton, if any.  Note that this schematic name is considered a
+    /// comment, and does not constitute a formal property value.
+    pub schematic_name: String,
+    /// The "joining type" of this codepoint.
+    pub joining_type: String,
```
Comment (BurntSushi) [inline ucd-parse/src/arabic_shaping.rs]: Hmm, looking at `ArabicShaping.txt`, it almost looks like this might be better defined as an enum?

**Review by BurntSushi — COMMENTED** [review]
Thanks! Just have a small question about whether the joining type should be an enum or not.

File: ucd-parse/src/arabic_shaping.rs (hunk 1 above)
Comment (wezm) [reply ucd-parse/src/arabic_shaping.rs]: Good idea, I've done that.

**Review by BurntSushi — APPROVED** [review]
Great, thanks!

**Pull request merged**